model/networks.py

```diff
@@ -285,10 +285,7 @@ def forward(self, f, b, mask=None):
             if self.use_cuda:
                 escape_NaN = escape_NaN.cuda()
             wi = wi[0]  # [L, C, k, k]
-            max_wi = torch.max(torch.sqrt(reduce_sum(torch.pow(wi, 2),
-                                                     axis=[1, 2, 3],
-                                                     keepdim=True)),
-                               escape_NaN)
+            max_wi = torch.sqrt(reduce_sum(torch.pow(wi, 2) + escape_NaN, axis=[1, 2, 3], keepdim=True))
```

Comment:
Did you forget to add `torch.max` here?

Reply:
I don't think torch.max is needed here. torch.pow(wi, 2) + escape_NaN makes torch.sqrt stable.
The old torch.max way is unstable. I think the reason is when wi is very small, torch.sqrt return NaN. 
And torch.max can not compare NaN with another number.
My training fails here a lot of times.

Reply:
Yes, you are right.

Reply:
Could you post some simple examples that `torch.sqrt` may return NaN?

I tried several small numbers with `torch.sqrt` but they don't return NaN. `torch.sqrt` may return NaN when it meets the negative numbers. I am curious about why there are negative numbers.

Reply:
I forget one thing. The NaN happened in backpropagation period. Because the derivative of sqrt is 1/2(1/sqrt). If the number is very small. It's inverse will be very large, that is NaN.

Reply:
```python
In [29]: a = torch.tensor(1e-50, requires_grad=True)

In [30]: q = torch.sqrt(a)

In [31]: q.backward()

In [32]: a.grad
Out[32]: tensor(inf)
```

That's right. Here is an example. When there is more computation with `inf`, we may get `nan` here.